homework_weekend7/src/task2/UniqueStackInt.java

```diff
@@ -0,0 +1,137 @@
+package task2;
+
+import java.util.EmptyStackException;
+
+/**
+ * Represents an implementation of stack of integers logic, with condition:
+ * elements can not be duplicated.
+ * 
+ * @author ajla
+ *
+ */
+public class UniqueStackInt {
+
+	private Node head;
+
+	/**
+	 * Adds element at the end of the stack.
+	 * 
+	 * @param elem
+	 * @return
+	 */
+	public int push(int elem) {
+		if (head == null) {
+			head = new Node(elem);
+			return elem;
+		}
+
+		Node tmp = head;
+
+		while (tmp.getNext() != null) {
+			if (tmp.getValue() == elem) {
+				System.out
+						.println("Desired element already exists and could not be pushed.");
+			} else {
+				tmp = tmp.getNext();
+			}
+		}
+
+		tmp = new Node(elem);
+		tmp.setNext(head);
+		head = tmp;
+		return elem;
+	}
+
+	/**
+	 * Removes last added element from the stack.
+	 * 
+	 * @return
+	 * @throws EmptyStackException
+	 */
+	public int pop() throws EmptyStackException {
+		if (head == null)
+			throw new EmptyStackException();
+		head = head.getNext();
```

Comment:
You should take value from current head before getting rid of it since as it is now, second element will be popped instead.